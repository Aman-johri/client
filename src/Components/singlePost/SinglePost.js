import React from 'react'
import "./singlePost.css";
import axios from 'axios'
import { connect } from 'react-redux';
import {deleteData, getData , updateData} from '../../redux/actions/PostActions'

class SinglePost extends React.Component {
  constructor(props){
    super(props);
    this.state = {
     post:this.props.Postdata,
     update:false
  }
  this.handleClick = this.handleClick.bind(this);
  this.handleClickUpdate = this.handleClickUpdate.bind(this);
}

  handleClick = e => {
    e.preventDefault();
    this.setState({
      update:true
    });
  }

  handleClickUpdate = e => {
    e.preventDefault();
    const postId = window.location.href.split("/")[4];
    const post = {
      title: this.state.title,
      desc: this.state.desc,
    };
    console.log(post);
    this.props.updateData(postId,post);
    window.alert("Congratulation! your post has been updated...");
    window.location.reload();
  }


  componentDidMount(){
    const postId = window.location.href.split("/")[4];
    this.props.getData(postId);
  }
  
   render(){
    return (
      <div className='singlePost'>
        {
            this.props.Postdata.map(post => (
              <div className='singlePostWrapper'>
              <img
              className="singlePostImg"
              src="https://images.pexels.com/photos/6685428/pexels-photo-6685428.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
              alt=""
            />
            { this.state.update ? <input type="text" id='title' 
             className="singlePostTitleInput" placeholder={post.title} value={this.props.Postdata.title} onChange={(e) => this.setState({title:e.target.value})}/> :
              <h1 className='singlePostTitle'>
                {post.title}
              <div className='singlePostEdit'>
              <i className='singlePostIcon fa fa-edit' onClick={this. handleClick} ></i>
              <i className='singlePostIcon fa fa-trash' onClick={deleteData(post._id)}></i>
              </div>
              </h1>
            }
            <div className='singlePostInfo'>
                <span className='singlePostAuthor'>Author: <b>Aman</b></span>
                <span className='singlePostDate'>{new Date(post.createdAt).toDateString()}</span>
            </div>
            {
              this.state.update ? <textarea className="singlePostDescInput" id="desc" placeholder={post.desc} onChange={(e) => this.setState({desc:e.target.value})}></textarea> :
              <p className='singlePostDesc'>
              {post.desc}
              </p>
            }
            <button className='singlePostBtn' onClick={this.handleClickUpdate} >Update</button>
            </div>
            ))
          }
      </div>
    )
  }
}

const mapStateToProps = (state) => {

  console.log(state.data);
  return {Postdata: state.data}

}

const mapDispatchToProps = (dispatch) => {
  return {
     deleteData: (postId) => dispatch(deleteData(postId)),
     getData: (postId) => dispatch(getData(postId)),
     updateData: (postId, post) => dispatch(updateData(postId, post))

  }
}


export default connect(mapStateToProps,mapDispatchToProps)(SinglePost);