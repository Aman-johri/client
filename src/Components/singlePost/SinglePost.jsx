import React from 'react'
import "./singlePost.css"

export default function SinglePost() {
  return (
    <div className='singlePost'>
        <div className='singlePostWrapper'>
        <img
          className="singlePostImg"
          src="https://images.pexels.com/photos/6685428/pexels-photo-6685428.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
          alt=""
        />
        <h1 className='singlePostTitle'>
            Lorem ipsum dolor sit amet.
        <div className='singlePostEdit'>
            <i className='singlePostIcon fa fa-edit'></i>
            <i className='singlePostIcon fa fa-trash'></i>
        </div>
        </h1>
        <div className='singlePostInfo'>
            <span className='singlePostAuthor'>Author: <b>Aman</b></span>
            <span className='singlePostDate'>1 hour ago</span>
        </div>
        <p className='singlePostDesc'>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Obcaecati, 
            aliquid debitis. Officiis eligendi nesciunt eaque, 
            aperiam aut enim reiciendis culpa beatae minus amet dolor temporibus expedita cupiditate numquam,
            quam quod commodi nemo eos nam in consequatur quis esse? Nulla voluptatem omnis assumenda debitis minus incidunt, 
            error temporibus veritatis necessitatibus nostrum.
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Obcaecati, 
            aliquid debitis. Officiis eligendi nesciunt eaque, 
            aperiam aut enim reiciendis culpa beatae minus amet dolor temporibus expedita cupiditate numquam,
            quam quod commodi nemo eos nam in consequatur quis esse? Nulla voluptatem omnis assumenda debitis minus incidunt, 
            error temporibus veritatis necessitatibus nostrum.
        </p>
        </div>
    </div>
  )
}
