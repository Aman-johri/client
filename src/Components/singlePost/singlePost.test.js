import SinglePost from "./SinglePost";
import {shallow,configure} from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });


describe("Testing", () => {
    it("snapshot test for singlepost", () => {
        const component = shallow(<SinglePost post={{ _id: "1234456767", title: "Aman Johri", desc:"Hello, I am Aman Johri", createdAt: "Sat Apr 02 2022"}} />);
        console.log(component);
        const tree = component.debug();
        expect(tree).toMatchSnapshot();
    });
});