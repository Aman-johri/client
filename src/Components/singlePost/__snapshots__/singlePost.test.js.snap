// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot test for singlepost 1`] = `
"<Fragment>
  <ForwardRef(Dialog) open={[undefined]} fullScreen={true}>
    <ForwardRef(AppBar) sx={{...}}>
      <ForwardRef(Toolbar)>
        <ForwardRef(IconButton) edge=\\"start\\" color=\\"inherit\\" aria-label=\\"close\\">
          <Memo(ForwardRef(CloseIcon)) onClick={[Function (anonymous)]} />
        </ForwardRef(IconButton)>
        <ForwardRef(Typography) sx={{...}} variant=\\"h6\\" component=\\"div\\">
          View Post Details or Update Post
        </ForwardRef(Typography)>
        <button className=\\"singlePostBtn\\" hidden={true} onClick={[Function (anonymous)]} id=\\"updatebtn\\">
          Update
        </button>
      </ForwardRef(Toolbar)>
    </ForwardRef(AppBar)>
    <div className=\\"singlePost\\">
      <div className=\\"singlePostWrapper\\">
        <img className=\\"singlePostImg\\" src=\\"https://images.pexels.com/photos/6685428/pexels-photo-6685428.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500\\" alt=\\"\\" />
        <h1 className=\\"singlePostTitle\\">
          Aman Johri
          <div className=\\"singlePostEdit\\">
            <i className=\\"singlePostIcon fa fa-edit\\" id=\\"edit\\" onClick={[Function (anonymous)]} />
          </div>
        </h1>
        <div className=\\"singlePostInfo\\">
          <span className=\\"singlePostAuthor\\">
            Author: 
            <b>
              Aman
            </b>
          </span>
          <span className=\\"singlePostDate\\">
            Sat Apr 02 2022
          </span>
        </div>
        <p className=\\"singlePostDesc\\">
          Hello, I am Aman Johri
        </p>
      </div>
    </div>
  </ForwardRef(Dialog)>
  <ForwardRef(Dialog) open={false} aria-labelledby=\\"alert-dialog-title\\" aria-describedby=\\"alert-dialog-description\\">
    <ForwardRef(DialogTitle) id=\\"alert-dialog-title\\">
      Are you sure you don&#39;t want to edit this post?
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogContent)>
      <ForwardRef(DialogContentText) id=\\"alert-dialog-description\\">
        If you click yes, you will be redirected to the home page.
      </ForwardRef(DialogContentText)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button) onClick={[Function (anonymous)]}>
        No
      </ForwardRef(Button)>
      <ForwardRef(Button) onClick={[Function (anonymous)]}>
        Yes
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
  <ForwardRef(Dialog) open={false} aria-labelledby=\\"alert-dialog-title\\" aria-describedby=\\"alert-dialog-description\\">
    <ForwardRef(DialogTitle) id=\\"alert-dialog-title\\">
      Are you sure you want to update your post?
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogContent)>
      <ForwardRef(DialogContentText) id=\\"alert-dialog-description\\">
        If you click yes, your previous data will be replaced by the new data.
      </ForwardRef(DialogContentText)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button) onClick={[Function (anonymous)]}>
        No
      </ForwardRef(Button)>
      <ForwardRef(Button) onClick={[Function (anonymous)]} id=\\"supportyes\\">
        Yes
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</Fragment>"
`;
